### DISCUSSION TITLE
Split components

### DISCUSSION THREAD

**Discussion by grepic** [twbs/bootstrap#37168] [Q&A] [answered]
Hi, im developing special website and i need to separate all source codes. I cant use bootstrap.min.js but what i need is split it to separete files like. collapse.js and collapse.css etc. I found this but it doesnt work if i will include collapse.js...so is there any way how to split it on componets? I dont want to use all source codes just few. Thanks a lot for some tips/ideas. 
![image](https://user-images.githubusercontent.com/33124105/190920104-89e2d44e-1985-4196-89d9-6ca86bbb36b8.png)

**Comment by Edditoria** [comment, marked as answer]
Consume the files in https://github.com/twbs/bootstrap/tree/main/js/dist ?

Or, in your JS code, `import { Collapse } from 'bootstrap/dist/js/bootstrap/bootstrap.esm.js';`?

As told in Bootstrap [docs][docs], it is quite complicated if you do it yourself. And, since you are developing website, I would use a bundler to resolve NPM packages, tree-shaking and distribute JS files for browser compatibility , e.g. Rollup, Webpack. Example:

```js
import { Collapse } from 'bootstrap'; // Type of Collapse is a class.
// You code...

// Or simply
export default Collapse
```

[docs]: https://getbootstrap.com/docs/5.2/getting-started/javascript/#using-bootstrap-as-a-module
